python/opensky_api.py

```diff
@@ -336,4 +415,110 @@ def get_flighs_from_interval(self, begin: int, end: int) -> list[FlightData]:
+        )
+
+        if states_json is not None:
+            flights_list = [FlightData(list(entry.values())) for entry in states_json]
+            return flights_list
+        return []
+
+    def get_arrivals_by_airport(self, airport: str, begin: int, end: int) -> list[FlightData]:
+        """
+        Retrieve flights for a certain airport which arrived within a given time interval [begin, end].
+        :param airport: ICAO identier for the airport
+        :param begin: Start of time interval to retrieve flights for as Unix time (seconds since epoch)
+        :param end: End of time interval to retrieve flights for as Unix time (seconds since epoch)
+        :return: list of FlightData objects.
+
+        """
+        if begin > end:
+            raise ValueError("The end parameter must be greater than begin")
+        if end - begin > 604800:
+            raise ValueError("The time interval must be smaller than 7 days")
+        if not self._check_rate_limit(0, 1, self.get_arrivals_by_airport):
+            logger.debug("Blocking request due to rate limit")
+            return []
+
+        params = {"airport": airport, "begin": begin, "end": end}
+        states_json = self._get_json(
+            "/flights/arrival", self.get_arrivals_by_airport, params=params
+        )
+
+        if states_json is not None:
+            flights_list = [FlightData(list(entry.values())) for entry in states_json]
+            return flights_list
+        return []
+
+    def get_departures_by_airport(self, airport: str, begin: int, end: int) -> list[FlightData]:
+        """
+        Retrieve flights for a certain airport which arrived within a given time interval [begin, end].
+        :param airport: ICAO identier for the airport
+        :param begin: Start of time interval to retrieve flights for as Unix time (seconds since epoch)
+        :param end: End of time interval to retrieve flights for as Unix time (seconds since epoch)
+        :return: list of FlightData objects.
+
+        """
+        if begin > end:
+            raise ValueError("The end parameter must be greater than begin")
+        if end - begin > 604800:
+            raise ValueError("The time interval must be smaller than 7 days")
+        if not self._check_rate_limit(0, 1, self.get_departures_by_airport):
+            logger.debug("Blocking request due to rate limit")
+            return []
+
+        params = {"airport": airport, "begin": begin, "end": end}
+        states_json = self._get_json(
+            "/flights/departure", self.get_departures_by_airport, params=params
+        )
+
+        if states_json is not None:
+            flights_list = [FlightData(list(entry.values())) for entry in states_json]
+            return flights_list
+        return []
+
+    def get_track_by_aircraft(self, icao24: str, t: int = 0) -> FlightTrack:
+        """
+        Retrieve flights for a certain airport which arrived within a given time interval [begin, end].
```

Comment:
The docstring does not fit. It is not about an airport.

Reply:
Done